apps/rush-lib/src/cli/actions/BaseRushAction.ts

```diff
@@ -11,19 +12,46 @@ import {
 import RushConfiguration from '../../data/RushConfiguration';
 import EventHooksManager from '../logic/EventHooksManager';
 
+export interface IRushCommandLineActionOptions extends ICommandLineActionOptions {
+  /**
+   * If true, no locking mechanism will be enforced when this action is run.
+   * Note this defaults to false (which is safer).
+   */
+  safeForSimultaneousRushProcesses?: boolean;
+}
+
 /**
  * The base Rush action that all Rush actions should extend.
  */
 export abstract class BaseRushAction extends CommandLineAction {
   private _rushConfiguration: RushConfiguration;
   private _eventHooksManager: EventHooksManager;
+  private _safeForSimultaneousRushProcesses: boolean;
 
-  constructor(options: ICommandLineActionOptions) {
+  constructor(options: IRushCommandLineActionOptions) {
     super(options);
+    this._safeForSimultaneousRushProcesses = !!options.safeForSimultaneousRushProcesses;
   }
 
   protected onExecute(): void {
     this._ensureEnvironment();
+
+    if (!this._safeForSimultaneousRushProcesses) {
+      const lockFilePath: string = path.join(
```

Comment:
>lockFilePath [](start = 12, length = 12)

Refactor to use the shared library?

Reply:
Done!

---
In reply to: [165526195](https://github.com/Microsoft/web-build-tools/pull/517#discussion_r165526195) [](ancestors = 165526195)